python_modules/dagster/dagster/_core/definitions/partition.py

```diff
@@ -1042,6 +1030,133 @@ def empty_subset(cls, partitions_def: PartitionsDefinition[T_str]) -> "Partition
         ...
 
 
+def get_default_partitions_subset_partitions_defs():
+    from .multi_dimensional_partitions import MultiPartitionsDefinition
+
+    DEFAULT_PARTITIONS_SUBSET_PARTITIONS_DEFS = tuple(
+        [
+            StaticPartitionsDefinition,
+            DynamicPartitionsDefinition,
+            MultiPartitionsDefinition,
+        ]
+    )
+    return DEFAULT_PARTITIONS_SUBSET_PARTITIONS_DEFS
+
+
+def get_time_window_partitions_subset_partitions_defs():
+    from .time_window_partitions import (
+        DailyPartitionsDefinition,
+        HourlyPartitionsDefinition,
+        MonthlyPartitionsDefinition,
+        TimeWindowPartitionsDefinition,
+        WeeklyPartitionsDefinition,
+    )
+
+    # TODO Check if all asset evaluations are constructed from external asset graphs
+    # If they are, we only need to check for TimeWindowPartitionsDefinition
+    TIME_WINDOW_PARTITIONS_SUBSET_PARTITIONS_DEFS = tuple(
```

Comment:
@OwenKephart do you know if it's possible for the `serialized_partitions_def_class_name` of a time partitions def to be anything other than `TimeWindowPartitionsDefinition`?

Could this be possible when using the asset reconciliation sensor on an old version of user code on cloud?

Reply:
Not sure if this is still a relevant question but at least at the moment I don't believe there's any way for that to be the case. e.g. if they're running the reconciliation sensor in old user code land, then the changes on this PR will not impact them as all of the computation will be happening in (non-updated) user code